features/steps/app_steps.rb

```diff
@@ -54,10 +111,66 @@
   end
 end
 
-#
-# Setting scenario mode
-#
+# No platform relevance
 
-When('I set the app to {string} mode') do |mode|
-  $scenario_mode = mode
+When('I clear the error queue') do
+  Maze::Server.errors.clear
+end
+
+def execute_command(action, scenario_name)
+  platform = Maze::Helper.get_current_platform
+  command = { action: action, scenario_name: scenario_name, scenario_mode: $scenario_mode }
+  case platform
+  when 'ios'
+    Maze::Server.commands.add command
+    trigger_app_command
+    $scenario_mode = nil
+    $reset_data = false
+    # Ensure fixture has read the command
+    count = 100
+    sleep 0.1 until Maze::Server.commands.remaining.empty? || (count -= 1) < 1
+    raise 'Test fixture did not GET /command' unless Maze::Server.commands.remaining.empty?
+  when 'macos'
+    Maze::Runner.environment['BUGSNAG_SCENARIO_ACTION'] = action.to_s
+    Maze::Runner.environment['BUGSNAG_SCENARIO_NAME'] = scenario_name.to_s
+    Maze::Runner.environment['BUGSNAG_SCENARIO_METADATA'] = $scenario_mode.to_s
+    Maze::Runner.environment['BUGSNAG_CLEAR_DATA'] = $reset_data ? 'true' : 'false'
+    $last_scenario = command
+    run_macos_app
+    $reset_data = false
+  else
+    raise "Unknown platform: #{platform}"
+  end
 end
+
+def trigger_app_command
+  platform = Maze::Helper.get_current_platform
+  case platform
+  when 'ios'
+    Maze.driver.click_element :execute_command
+  when 'macos'
+    run_macos_app
+  else
+    raise "Unknown platform: #{platform}"
+  end
+end
+
+def wait_for_true
+  max_attempts = 300
+  attempts = 0
+  assertion_passed = false
+  until (attempts >= max_attempts) || assertion_passed
+    attempts += 1
+    assertion_passed = yield
+    sleep 0.1
+  end
+  raise 'Assertion not passed in 30s' unless assertion_passed
+end
+
+def run_macos_app
+  Maze::Runner.kill_running_scripts if $reset_data
+  Maze::Runner.run_command("features/fixtures/macos/output/#{Maze.config.app}.app/Contents/MacOS/#{Maze.config.app}", blocking: false)
```

Comment:
Given we always pass the same app name, is it even worth making this configurable?

The steps are so highly coupled to the behaviour of the app, this "flexibility" seems like an illusion...